src/mmrelay/meshtastic_utils.py

```
@@ -519,6 +517,39 @@ def connect_meshtastic(passed_config=None, force_connect=False):
+            connection_timeout = DEFAULT_CONNECTION_TIMEOUT
+
+    # Pre-compute BLE interface signature outside retry loop to avoid mocking issues
+    ble_init_sig = None
+    if connection_type == CONNECTION_TYPE_BLE:
+        try:
+            ble_init_sig = inspect.signature(
+                meshtastic.ble_interface.BLEInterface.__init__
+            )
+        except (ValueError, TypeError, AttributeError):
+            # Use empty signature as fallback for mock environments
+            ble_init_sig = inspect.signature(lambda **kwargs: None)
```

Ruff (0.14.8)

Unused lambda argument: `kwargs`

(ARG005)